src/authorization/WebAclAuthorizer.ts

```diff
@@ -56,9 +73,12 @@ export class WebAclAuthorizer extends Authorizer {
     const modes = (Object.keys(permissions) as (keyof PermissionSet)[]).filter((key): boolean => permissions[key]);
     this.logger.debug(`Checking if ${credentials.webId} has ${modes.join()} permissions for ${identifier.path}`);
 
+    // Reset the cache of vCard group files at the start of each interaction
+    this.groupsCache = {};
```

Comment:
Having gone through the code now, the issues of the above comment still stand, but I'm going to suggest a different solution. I would use a `KeyValueStorage` here. Specifically because such a storage can be wrapped in a `WrappedExpiringStorage` which allows temporary data. That would solve the comment about expiring data below. Config example: https://github.com/solid/community-server/blob/e1ed9c823e2943c2d71e9220f9dd086a0752bb9a/config/identity/handler/key-value/storage.json#L4-L9

Note that a WrappedExpiringStorage needs to be added to the list of Finalizers so it can be cleaned up once the server stops. The same config mentioned above also has this: https://github.com/solid/community-server/blob/e1ed9c823e2943c2d71e9220f9dd086a0752bb9a/config/identity/handler/key-value/storage.json#L10-L14

How the storages they are wrapped around are configured you can see in https://github.com/solid/community-server/tree/main/config/storage/key-value . Since you are storing `Store` objects though, this would have to be configured to always be a Memory storage (since we can't store JS objects in the ResourceStore), so its configuration does not have to happen in the folder above since it can't change. We could potentially change the cache to store the body of the `fetch` call instead of the Store, that way we could use the different storage, but that would be a more drastic change of this PR and not immediately required.

Reply:
Hi @joachimvh and @RubenVerborgh ,
I'm the co-author of this PR. We are trying to implement the cache for groupAccessChecker. Since the handle method of this class will be executed in parallel multiple times, we would need a locking mechanism for accessing the cache. Can you suggest us a way to do this? have you already implemented something like that in this project? Thank you very much

Reply:
Concurring with Ruben below, I don't think locking is necessary here. In the end this is also data that doesn't change once it is written to the cache, the only modification that happens is that it gets deleted at some point which will simply return undefined on a get. In the case of the memory map storage, objects get added or removed to a simple Map object, and in the case of the resource storage it goes through all the resource stores which includes a locking mechanism, so both of these are safe.

Reply:
Hi @joachim. I've forced push this branch some minutes ago.
Please, have a look at the `GroupAccessChecker.ts` file (in particular, at the `isMemberOfGroup` method). That method is called inside a "promiseAny" call, which means that potentially many instances of it could be executed in parallel.

What happens, when we try to use a cache inside that method, is that every instance reads the cache almost at the same time (discovering that it does not contain the required object) and so each of them proceeds by fetching that resource.

We thought about some locking mechanism because we wanted to prevent multiple fetches of the same resource, but maybe there's a different way to handle the cache that avoids this problem altogether... We don't know 🤷‍♂️...